ddl/column.go

```diff
@@ -472,8 +472,15 @@ func (w *worker) doModifyColumn(t *meta.Meta, job *model.Job, newCol *model.Colu
 
 // checkForNullValue ensure there are no null values of the column of this table.
 // `isDataTruncated` indicates whether the new field and the old field type are the same, in order to be compatible with mysql.
-func checkForNullValue(ctx sessionctx.Context, isDataTruncated bool, schema, table, oldCol, newCol model.CIStr) error {
-	sql := fmt.Sprintf("select 1 from `%s`.`%s` where `%s` is null limit 1;", schema.L, table.L, oldCol.L)
+func checkForNullValue(ctx sessionctx.Context, isDataTruncated bool, schema, table, newCol model.CIStr, oldCols ...*model.ColumnInfo) error {
+	colsStr := ""
+	for i, col := range oldCols {
+		colsStr += "`" + col.Name.L + "` is null"
+		if i != 0 {
+			colsStr += " or `" + col.Name.L + "` is null"
+		}
```

Comment:
Without else?
```go
		if i != 0 {
			colsStr += " or `" + col.Name.L + "` is null"
		} else {
			colsStr += "`" + col.Name.L + "` is null"
		}
```

Please add test for this.

Reply:
Good catch. Done.